ddtrace/opentelemetry/_span.py

```diff
@@ -168,8 +168,20 @@ def set_attribute(self, key, value):
 
     def add_event(self, name, attributes=None, timestamp=None):
         # type: (str, Optional[Attributes], Optional[int]) -> None
-        """NOOP - events are not yet supported"""
-        return
+        """Records an event"""
+        if not self.is_recording():
+            return
+
+        flattened_attrs = {}
+        for key, value in (attributes or {}).items():
+            # flatten the attributes and set them as tags
+            if is_sequence(value):
+                for k, v in flatten_key_value(key, value).items():
```

Comment:
I know the development of the spec has been fast, but we've specified in the latest RFC that we shouldn't flatten an array into its dot-notation. I started a new comment thread to raise the discussion in the document and we can come back here with the conclusion

Reply:
I will revert this change